src/Symfony/Component/Process/Process.php

```diff
@@ -468,7 +468,7 @@ public function getIncrementalErrorOutput()
      */
     public function getExitCode()
     {
-        if ($this->isSigchildEnabled() && !$this->enhanceSigchildCompatibility) {
+        if (!$this->enhanceSigchildCompatibility && $this->isSigchildEnabled()) {
```

Comment:
I'm against all these changes. IIUC, it's a small perf optim, but probably negligible and it makes it very difficult to merge branches between them.